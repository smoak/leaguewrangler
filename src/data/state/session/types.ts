export interface Session {
  readonly userToken: string;
}
