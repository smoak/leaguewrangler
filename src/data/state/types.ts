import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
