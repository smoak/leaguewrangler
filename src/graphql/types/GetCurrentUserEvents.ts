/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RsvpStatus } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetCurrentUserEvents
// ====================================================

export interface GetCurrentUserEvents_currentUser_events_team {
  __typename: "Team";
  shortName: string;
  photoThumbnailUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events_location {
  __typename: "Location";
  name: string;
  address: string | null;
  googleMapsUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events {
  __typename: "Event";
  eventId: number;
  title: string;
  endTimestamp: number | null;
  startTimestamp: number | null;
  viewerRsvpStatus: RsvpStatus;
  team: GetCurrentUserEvents_currentUser_events_team;
  location: GetCurrentUserEvents_currentUser_events_location | null;
}

export interface GetCurrentUserEvents_currentUser {
  __typename: "User";
  events: (GetCurrentUserEvents_currentUser_events | null)[];
}

export interface GetCurrentUserEvents {
  /**
   * Gets the current user
   */
  currentUser: GetCurrentUserEvents_currentUser;
}
