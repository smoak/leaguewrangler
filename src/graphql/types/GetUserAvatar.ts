/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetUserAvatar
// ====================================================

export interface GetUserAvatar_currentUser {
  __typename: "User";
  profilePhotoThumbnailUrl: string;
}

export interface GetUserAvatar {
  /**
   * Gets the current user
   */
  currentUser: GetUserAvatar_currentUser;
}
