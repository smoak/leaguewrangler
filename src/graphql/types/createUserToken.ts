import * as Types from './globalTypes';

export type CreateUserTokenMutationVariables = Types.Exact<{
  username: Types.Scalars['String'];
  password: Types.Scalars['String'];
}>;


export type CreateUserTokenMutation = { __typename?: 'Mutation', createUserToken?: { __typename?: 'UserToken', token: string } | null };
