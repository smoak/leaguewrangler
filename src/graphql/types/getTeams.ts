import * as Types from './globalTypes';

export type GetTeamsQueryVariables = Types.Exact<{ [key: string]: never; }>;


export type GetTeamsQuery = { __typename?: 'Query', teams: Array<{ __typename?: 'Team', name: string, id: number } | null> };
