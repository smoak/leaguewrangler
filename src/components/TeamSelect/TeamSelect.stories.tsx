import { MockedProvider, MockedResponse } from '@apollo/react-testing';
import { storiesOf } from '@storybook/react';
import * as React from 'react';

import getTeams from '../../graphql/queries/getTeams';

import TeamSelect from './TeamSelect';

const mocksWithTeams: ReadonlyArray<MockedResponse> = [
  {
    request: {
      query: getTeams,
    },
    result: {
      data: {
        teams: [
          {
            teamId: 1,
            name: 'Team 1',
          },
          {
            teamId: 2,
            name: 'Team 2',
          },
        ],
      },
    },
  },
];

storiesOf('TeamSelect', module).add('with teams', () => (
  <MockedProvider mocks={mocksWithTeams} addTypename={false}>
    <TeamSelect />
  </MockedProvider>
));
