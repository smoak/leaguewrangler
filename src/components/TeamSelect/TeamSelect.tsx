import { useQuery } from '@apollo/client';
import { CircularProgress, FormControl, MenuItem, Select } from '@mui/material';
import { FC } from 'react';

import getTeamsQuery from '../../graphql/queries/getTeams';
import { GetTeamsQuery } from '../../graphql/types/getTeams';

function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
  return value !== null && value !== undefined;
}

const TeamSelect: FC = () => {
  const { loading, data, error } = useQuery<GetTeamsQuery>(getTeamsQuery);

  if (loading) {
    return <CircularProgress color="secondary" />;
  }

  if (!data || error) {
    return null;
  }

  const { teams } = data;

  const firstTeam = teams[0];

  return (
    <FormControl>
      <Select value={firstTeam ? firstTeam.id : ''} name="team">
        {teams.filter(notEmpty).map((team) => (
          <MenuItem key={team.id} value={team.id}>
            {team.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default TeamSelect;
