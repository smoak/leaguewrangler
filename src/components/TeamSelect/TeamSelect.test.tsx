import { useQuery } from '@apollo/react-hooks';
import { ShallowWrapper, shallow } from 'enzyme';
import { GetTeams } from 'graphql/types/GetTeams';
import React from 'react';

import TeamSelect from './TeamSelect';

describe('TeamSelect', () => {
  let component: ShallowWrapper;

  describe('when the data is loading', () => {
    beforeEach(() => {
      (useQuery as jest.Mock).mockReturnValue({ loading: true });
      component = shallow(<TeamSelect />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the server returns an error', () => {
    beforeEach(() => {
      (useQuery as jest.Mock).mockReturnValue({ loading: false, data: {}, error: 'nope' });
      component = shallow(<TeamSelect />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the server returns data', () => {
    describe('and the user belongs to no teams', () => {
      beforeEach(() => {
        const teams: GetTeams['teams'] = [];
        (useQuery as jest.Mock).mockReturnValue({ loading: false, data: { teams }, error: undefined });

        component = shallow(<TeamSelect />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('and the user belongs to a single team', () => {
      beforeEach(() => {
        const teams: GetTeams['teams'] = [{ name: 'Test Team', id: 1, __typename: 'Team' }];
        (useQuery as jest.Mock).mockReturnValue({ loading: false, data: { teams }, error: undefined });

        component = shallow(<TeamSelect />);
      });

      it('renders as expected', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });
});
