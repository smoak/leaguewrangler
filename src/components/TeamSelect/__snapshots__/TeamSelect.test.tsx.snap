// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSelect when the data is loading renders as expected 1`] = `
<WithStyles(CircularProgress)
  color="secondary"
/>
`;

exports[`TeamSelect when the server returns an error renders as expected 1`] = `""`;

exports[`TeamSelect when the server returns data and the user belongs to a single team renders as expected 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(Select))
    name="team"
    value={1}
  >
    <WithStyles(MenuItem)
      key="1"
      value={1}
    >
      Test Team
    </WithStyles(MenuItem)>
  </WithStyles(WithFormControlContext(Select))>
</WithStyles(FormControl)>
`;

exports[`TeamSelect when the server returns data and the user belongs to no teams renders as expected 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(Select))
    name="team"
    value=""
  />
</WithStyles(FormControl)>
`;
